### PR TITLE
Fix snapshot restore

### PR DESCRIPTION
**Pull request by dincho**
In scope of https://www.pivotaltracker.com/story/show/167595374

This is _soft_ backward compatibility break, that is, an environment that lack `snapshot_filename` tag will not be restored from a snapshot, however the environment should provision and sync just fine.

### CODE REVIEW

File: ansible/mnesia_snapshot.yml [hunk 1]
```diff
@@ -21,27 +26,35 @@
...
+
+    - name: Archive Mnesia Directory (with additional storage)
       archive:
-        format: "{{ format }}"
-        path: "{{ project_root + '/' if additional_storage is not defined else '' }}{{ node_config.chain.db_path }}"
-        dest: /tmp/{{ backup_archive_filename }}
+        format: gz
+        path: "{{ project_root }}/{{ node_config.chain.db_path }}"
+        dest: /tmp/{{ snapshot_filename }}
+      when: additional_storage is defined
```
Comment (venimus) [inline ansible/mnesia_snapshot.yml]: and additional_storage

File: ansible/mnesia_snapshot.yml [hunk 2]
```diff
@@ -21,27 +26,35 @@
 
     - meta: flush_handlers
 
-    - name: Archive Mnesia Directory
+    - name: Archive Mnesia Directory (no additional storage)
+      archive:
+        format: gz
+        path: "{{ project_root }}/{{ node_config.chain.db_path }}"
+        dest: /tmp/{{ snapshot_filename }}
+      when: additional_storage is not defined
```
Comment (venimus) [inline ansible/mnesia_snapshot.yml]: or not addtional_storage

File: ansible/mnesia_snapshot.yml (hunk 1 above)
Comment (dincho) [reply ansible/mnesia_snapshot.yml]: fixed in 9c0cf35

File: ansible/mnesia_snapshot.yml (hunk 2 above)
Comment (dincho) [reply ansible/mnesia_snapshot.yml]: fixed in 9c0cf35

File: ansible/mnesia_snapshot.yml [hunk 3]
```diff
@@ -21,39 +27,51 @@
...
         mode: put
         aws_access_key: "{{ lookup('env','AWS_ACCESS_KEY_ID') }}"
         aws_secret_key: "{{ lookup('env','AWS_SECRET_ACCESS_KEY') }}"
         security_token: "{{ lookup('env','AWS_SESSION_TOKEN') }}"
 
-    - name: Remove Mnesia backups
+    - name: Delete local Mnesia snapshots ({{ snapshots_dir }})
       file:
-        path:  "{{ item }}"
+        path: "{{ snapshots_dir }}"
```
Comment (lucafavatella) [inline ansible/mnesia_snapshot.yml]: All? or snapshot_path?

Comment (dincho) [reply ansible/mnesia_snapshot.yml]: Yes, it is meant to delete all snapshots in that directory. The use-case is if something fail in the middle, next time it's cleanedup.

**Pull request merged**